internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -273,6 +295,20 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 	}
 	u.log.Debugf("detected used flavor: %q", detectedFlavor)
 	unpackRes, err := u.unpack(version, archivePath, paths.Data(), detectedFlavor)
+
+	// If VersionedHome is empty then unpack has not started unpacking the
+	// archive yet. There's nothing to clean up. Return the error.
+	if unpackRes.VersionedHome == "" {
```

Comment:
@pchila is still field always present? i recall it can be empty in some cases, but maybe it's just something else i had in mind